### PR TITLE
frink: 2023-01-31 -> 2023-05-22

### PR DESCRIPTION
**Pull request by fehrenbach**
###### Description of changes

Upstream version changes: https://frinklang.org/whatsnew.html

This includes the changes for rlwrap helper files, see #230318.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/frink/default.nix [hunk 1]
```diff
@@ -27,9 +27,18 @@ stdenv.mkDerivation rec {
 
     cp ${src} $out/lib/frink.jar
 
+    # Generate rlwrap helper files.
+    # See https://frinklang.org/fsp/colorize.fsp?f=listUnits.frink
+    # and https://frinklang.org/fsp/colorize.fsp?f=listFunctions.frink
+    ${jdk}/bin/java -classpath "$out/lib/frink.jar" frink.gui.FrinkStarter -e 'joinln[lexicalSort[units[]]]' > $out/lib/unitnames.txt
```
Comment (wegank) [inline pkgs/development/tools/frink/default.nix:33]:
```suggestion
    java -classpath "$out/lib/frink.jar" frink.gui.FrinkStarter -e 'joinln[lexicalSort[units[]]]' > $out/lib/unitnames.txt
```

With `jdk` put in `nativeBuildInputs` additionally.

File: pkgs/development/tools/frink/default.nix [hunk 2]
```diff
@@ -27,9 +27,18 @@ stdenv.mkDerivation rec {
...
+    ${jdk}/bin/java -classpath "$out/lib/frink.jar" frink.gui.FrinkStarter -e 'joinln[map[{|f|
```
Comment (fehrenbach) [inline pkgs/development/tools/frink/default.nix:34]:
```suggestion
    java -classpath "$out/lib/frink.jar" frink.gui.FrinkStarter -e 'joinln[map[{|f|
```

File: pkgs/development/tools/frink/default.nix [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
 
   src = fetchurl {
     # Upstream does not provide versioned download links
-    url = "https://web.archive.org/web/20230202134810/https://frinklang.org/frinkjar/frink.jar";
-    sha256 = "sha256-xs1FQvFPgeAxscAiwBBP8N8aYe0OlsYbH/vbzzCbYZc=";
+    url = "https://web.archive.org/web/20230526123219/https://frinklang.org/frinkjar/frink.jar";
+    sha256 = "sha256-IgINJvt9G5f1HELKhV5BHIu9NoA8STDqNg/dVTFzK0Y=";
   };
 
   dontUnpack = true;
```
Comment (fehrenbach) [inline pkgs/development/tools/frink/default.nix:19]:
```suggestion
  dontUnpack = true;
  
  nativeBuildInputs = [ jdk ];
```

File: pkgs/development/tools/frink/default.nix (hunk 1 above)
Comment (fehrenbach) [reply pkgs/development/tools/frink/default.nix]: Done (and squashed), thanks.

**Review by wegank — APPROVED** [review]

**Pull request merged**